File: fix_props.py
with open("src/components/SettingsView.tsx", "r") as f:
    content = f.read()

if "holidays: HolidayRecord[];" not in content:
    content = content.replace("schedules: WorkSchedule[];", "schedules: WorkSchedule[];\n  holidays?: any[];")
    content = content.replace("schedules,", "schedules,\n  holidays,")
    with open("src/components/SettingsView.tsx", "w") as f:
        f.write(content)

with open("src/components/LaporanView.tsx", "r") as f:
    content = f.read()

if "holidays?: any[];" not in content:
    content = content.replace("schedules?: any[];", "schedules?: any[];\n  holidays?: any[];")
    content = content.replace("schedules,", "schedules,\n  holidays = [],")
    with open("src/components/LaporanView.tsx", "w") as f:
        f.write(content)

File: fix_props.ts
import fs from 'fs';
let content = fs.readFileSync('src/components/LaporanView.tsx', 'utf-8');
content = content.replace(
  "slipUbarList: SlipUbarRecord[];\n}",
  "slipUbarList: SlipUbarRecord[];\n  schedules?: any[];\n}"
);
content = content.replace(
  "slipUbarList,\n}) => {",
  "slipUbarList,\n  schedules,\n}) => {"
);
fs.writeFileSync('src/components/LaporanView.tsx', content);
